src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Mailbox.java

```diff
@@ -17,24 +19,28 @@ public final class Mailbox extends AggregateRoot<MailboxId> {
    */
   private final Address address;
 
-  private List<Email> emails;
+  private final Map<Email, List<Label>> emails;
   private final MailboxType type;
 
-  private Mailbox(MailboxId id, Address address, List<Email> emails, MailboxType type) {
+  private Mailbox(MailboxId id, Address address, Map<Email, List<Label>> emails, MailboxType type) {
     super(id);
     if (type == null) {
-      type = MailboxType.READ;
+      type = MailboxType.INBOX;
     }
     this.emails = emails;
     this.address = address;
     this.type = type;
   }
 
-  public static Mailbox create(Address address, List<Email> emails, MailboxType type) {
+  public static Mailbox create(Address address, Map<Email, List<Label>> emails, MailboxType type) {
```

Comment:
maybe it is better to use a set of labels to avoid duplicate labels

Reply:
nvm I thought that a user can create their own labels, but we already define labels in the enum

Reply:
>  I thought that a user can create their own labels

Depending on the remaining time, we could totally do that. :)

Reply:
> nvm I thought that a user can create their own labels, but we already define labels in the enum

also, only the mailbox knows if an email was labeled as READ or UNREAD. This means we have to use a Map to identify whether an email is read or unread